rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JDataset.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+import org.apache.commons.rdf.api.Dataset;
+import org.apache.commons.rdf.api.Quad;
+
+
+/**
+ * Marker interface for RDF4J implementations of Dataset.
+ * 
+ */
+public interface RDF4JDataset extends Dataset, RDF4JGraphLike<Quad> {
```

Comment:
[GraphLike](http://stain.github.io/incubator-commonsrdf/COMMONSRDF-35/org/apache/commons/rdf/api/GraphLike.html) is the common interface between `Graph` and `Dataset` - operating over `TripleLike` entries.

This is mainly useful for general code that can deal with both graph and dataset (e.g. a parser/writer that iterates over `TripleLike`) - a `Graph` and `Dataset` would have different equality semantics on operations like `.contains()` and `remove()`.

We might not need a corresponding [RDF4JGraphLike](http://stain.github.io/incubator-commonsrdf/COMMONSRDF-35/org/apache/commons/rdf/rdf4j/RDF4JGraphLike.html) - it  exposes `.asModel()` and `asRepository()`.

Internally in `rdf4j/` there's an abstract class [AbstractRepositoryGraphLike](http://stain.github.io/incubator-commonsrdf/COMMONSRDF-35/org/apache/commons/rdf/rdf4j/impl/AbstractRepositoryGraphLike.html) which shares the code of [RepositoryDatasetImpl](http://stain.github.io/incubator-commonsrdf/COMMONSRDF-35/org/apache/commons/rdf/rdf4j/impl/RepositoryDatasetImpl.html) (implementing `Dataset`) and [RepositoryGraphImpl](http://stain.github.io/incubator-commonsrdf/COMMONSRDF-35/org/apache/commons/rdf/rdf4j/impl/RepositoryGraphImpl.html) (implementing `Graph`).